scripts/ci/get_python_version.py

```diff
@@ -0,0 +1,132 @@
+            if ado_branch_last_commit == '$(System.PullRequest.SourceCommitId)':
+                # default value if ADO_PULL_REQUEST_LATEST_COMMIT not set in ADO
+                continue
+            elif ado_target_branch == '$(System.PullRequest.TargetBranch)':
+                # default value if ADO_PULL_REQUEST_TARGET_BRANCH not set in ADO
+                continue
+            else:
+                cmd = cmd_tpl.format(commit_start=ado_target_branch, commit_end=ado_branch_last_commit,
+                                     code_dir=src_d_full)
+                if not check_output(shlex.split(cmd)):
+                    continue
+
+        # Find the package and check it has tests
+        test_dir = os.path.isdir(os.path.join(src_d_full, pkg_name, 'tests'))
+        if pkg_name and test_dir:
+            # [('azext_healthcareapis', '/mnt/vss/_work/1/s/src/healthcareapis')]
+            all_tests.append((pkg_name, src_d_full))
+            # azext_healthcareapis -> healthcareapis
+            _, extension_name = pkg_name.split('_', 1)
+            logger.info(f'ado_branch_last_commit: {ado_branch_last_commit}, '
+                        f'ado_target_branch: {ado_target_branch}, '
+                        f'detect which extension need to test: {all_tests}.')
+            return pkg_name, extension_name
+    else:
+        logger.error(f'can not any test')
+        with open('python_version.txt', 'w') as f:
+            f.write('false')
+        sys.exit(0)
+
+
+def get_min_version(pkg_name, extension_name):
+    try:
+        with open(os.path.join(SRC_PATH, extension_name, pkg_name, 'azext_metadata.json'), 'r') as f:
+            min_version = json.load(f)['azext.minCliCoreVersion']
+            logger.info(f'find minCliCoreVersion: {min_version}')
+            return min_version
+    except Exception as e:
+        logger.error(f'can not get minCliCoreVersion: {e}')
+        with open('python_version.txt', 'w') as f:
+            f.write('false')
+        sys.exit(0)
+
+
+def get_python_version(min_version):
+    logger.info(f'checkout to minCliCoreVersion: {min_version}')
+    az_min_version = 'azure-cli-' + min_version
+    azure_cli_path = os.path.join(os.path.abspath(os.path.dirname(os.path.dirname(SRC_PATH))), 'azure-cli')
+    # checkout to min cli version
+    cmd = ['git', 'checkout', az_min_version]
+    run_command(cmd, check_return_code=True, cwd=azure_cli_path)
+    # display cli version
+    cmd = ['az', '--version']
+    run_command(cmd, check_return_code=True)
+    try:
+        core_path = os.path.join(azure_cli_path, 'src', 'azure-cli-core')
+        with open(os.path.join(core_path, 'setup.py'), 'r') as f:
+            for line in f.readlines():
+                if 'Programming Language :: Python :: ' in line:
+                    _, _, python_version = line.split(' :: ')
+                    python_version = python_version.split('\'')[0]
+        logger.info(f'return python verison: {python_version}')
```

Comment:
Do we only test the highest Python version which can support the `minCliCoreVersion` of azure-cli-core?
If so, if a PR contains incompatible contents for a lower Python version, but the lower Python version can also support the `minCliCoreVersion` of azure-cli-core which users may use it, is such such problem undetectable?

Reply:
That make sence.
Try to control the python version that needs to be executed through condition.
python3.10 will be converted to python3.1, so a separate job is required.
I have submitted a related issue in azure-pipelines: https://github.com/microsoft/azure-pipelines-tasks/issues/17323